packages/react-native-web/src/exports/Modal/ModalAnimation.js

```diff
@@ -0,0 +1,160 @@
+  const {
+    children,
+    style,
+    animated,
+    animationType,
+    visible,
+    onShow,
+    onDismiss
+  } = props;
+
+  const [isRendering, setIsRendering] = useState(false);
+
+  // Resolve the "actual" animation type by checking the (somewhat deprecated)
+  // animated prop against the animationType prop
+  const computedAnimationType = useMemo(() => {
+    if (!animationType) {
+      if (animated) {
+        return 'slide';
+      } else {
+        return 'none';
+      }
+    }
+
+    return animationType;
+  }, [animationType, animated]);
+
+  const isAnimated = computedAnimationType !== 'none';
+
+  // Wrap the callbacks so we don't have to worry about them being null
+  const onShowCallback = useCallback(() => { if (onShow) { onShow(); } }, [onShow]);
+  const onDismissCallback = useCallback(() => { if (onDismiss) { onDismiss(); } }, [onDismiss])
+
+  const animationEndCallback = useCallback(() => {
+    if (visible) {
+      // If animation completed and we're visible,
+      // fire off the onShow callback
+      onShowCallback();
+    } else {
+      // If animation completed and we're visible,
+      // fire off the onDismiss callback and stop rendering
+      setIsRendering(false);
+      onDismissCallback();
+    }
+  }, [onDismissCallback, onShowCallback, visible]);
+
+  // If the `visible` flag is changing we want to set the rendering flag to true
+  // before the animations ever will start
+  useEffect(() =>  {
+    if (visible) {
+      setIsRendering(true);
+    }
+
+    if (!isAnimated) {
+      // If !isAnimated we have to manually call `animationEndCallback` - if we
+      // don't it will never get called
+      animationEndCallback();
+    }
+  }, [isAnimated, visible, animationEndCallback]);
+
+  if (!isRendering) {
+    return null;
+  }
+
+  return createElement(
+    'div',
+    {
+      style: [style, getAnimationStyle(animationType, visible)],
+      onAnimationEnd: animationEndCallback,
+      children
+    }
+  );
+}
+
+const styles = StyleSheet.create({
+  animatedIn: {
+    animationDuration: `${ANIMATION_DURATION}ms`,
+    animationTimingFunction: 'ease-in'
+  },
+  animatedOut: {
+    pointerEvents: 'none',
+    animationDuration: `${ANIMATION_DURATION}ms`,
+    animationTimingFunction: 'ease-out'
+  },
+  fadeIn: {
+    opacity: 1,
+    animationKeyframes: {
+      '0%': { opacity: 0 },
+      '100%': { opacity: 1 }
+    }
+  },
+  fadeOut: {
+    opacity: 0,
+    animationKeyframes: {
+      '0%': { opacity: 1 },
+      '100%': { opacity: 0 }
+    }
+  },
+  slideIn: {
+    transform: [ { translateY: '0%' } ],
+    animationKeyframes: {
+      '0%': { transform: [ { translateY: '100%' } ] },
+      '100%': { transform: [ { translateY: '0%' } ] }
+    }
+  },
+  slideOut: {
+    transform: [ { translateY: '100%' } ],
+    animationKeyframes: {
+      '0%': { transform: [ { translateY: '0%' } ] },
+      '100%': { transform: [ { translateY: '100%' } ] }
+    }
+  },
```

Comment:
Did you try using `animationDirection:'reverse'` instead of defining the "out" animations?

Reply:
I did.  The problem I ran into with that was why I've included a `transform` outside of the keyframes.

At the end of the animation but before `OnAnimationEnd` fires there's a brief moment of the animation resetting to the `0%` - even when I was fiddling with the fill modes and other animation options.

With the `transform` added in the `slideOut` as `translateY(100%)` after the animation completes it will maintain the `100%` position.

For reference, I saw this on Firefox.